utils/stdx/src/lib.rs

```diff
@@ -98,31 +102,41 @@ pub fn as_chunks<const N: usize, T>(slice: &[T]) -> (&[[T; N]], &[T]) {
     (head, tail)
 }
 
-/// Like `as_chunks` but returns an error if there’s a remainder.
-pub fn as_chunks_exact<const N: usize, T>(slice: &[T]) -> Result<&[[T; N]], ()> {
+/// Chunks if `as_chunks` has no remainder.
+pub fn as_chunks_exact<const N: usize, T>(slice: &[T]) -> Option<&[[T; N]]> {
     let (chunks, remainder) = as_chunks(slice);
     if remainder.is_empty() {
-        Ok(chunks)
+        Some(chunks)
     } else {
-        Err(())
+        None
     }
 }
 
 #[test]
 fn test_as_chunks() {
     assert_eq!((&[[0, 1], [2, 3]][..], &[4][..]), as_chunks::<2, _>(&[0, 1, 2, 3, 4]));
-    assert_eq!(Ok(&[[0, 1], [2, 3]][..]), as_chunks_exact::<2, _>(&[0, 1, 2, 3]));
-    assert_eq!(Err(()), as_chunks_exact::<2, _>(&[0, 1, 2, 3, 4]));
+    assert_eq!(Some(&[[0, 1], [2, 3]][..]), as_chunks_exact::<2, _>(&[0, 1, 2, 3]));
+    assert_eq!(None, as_chunks_exact::<2, _>(&[0, 1, 2, 3, 4]));
 }
 
 /// Asserts, at compile time, that `S == A + B`.
 struct AssertEqSum<const S: usize, const A: usize, const B: usize>;
 impl<const S: usize, const A: usize, const B: usize> AssertEqSum<S, A, B> {
-    const OK: () = [()][A + B - S];
+    const OK: () = static_assert(S == A + B);
 }
 
 /// Asserts, at compile time, that `N` is non-zero.
 struct AssertNonZero<const N: usize>;
 impl<const N: usize> AssertNonZero<N> {
-    const OK: () = [()][if N == 0 { 1 } else { 0 }];
+    const OK: () = static_assert(N > 0);
+}
+
+const fn static_assert(ok: bool) {
+    // Array with a single unit element
+    let arr = [()];
+    // index == 0 ↔ ok == true
+    let index = !ok as usize;
+    // the following expression will fail with
+    // "index out of bounds" if ok is false
+    arr[index]
```

Comment:
Yeah that's still hacky but at least no longer obfuscated :D 

But I think we can do better! Apparently `const_panic` has been stable for a while, so a dependency to static_assertions is not even necessary. That is, you can just use regular `assert!` in a const fn. Example taken from https://internals.rust-lang.org/t/nicer-static-assertions/15986

```rust
const _: () = assert!(FOO == BAR, "assert_foo_equals_bar");
```

Reply:
that is awesome, thanks, updated!